### PR TITLE
MMCA-5046-update date formatter to allow searches this month

### PR DESCRIPTION
**Pull request by AdamCoops**

### CODE REVIEW

File: app/forms/mappings/SelectLocalDateFormatter.scala [hunk 1]
```diff
@@ -139,6 +141,20 @@ private[mappings] class SelectLocalDateFormatter(emptyMonthAndYearKey: String,
...
 
+  private def getDay(month: Int, year: Int) = {
+    val today = LocalDate.now
+    val todayMonth = today.getMonthValue
+    val todayYear = today.getYear
+    
+    val day: Int = (month, year) match {
+      case _ if useLastDayOfMonth => YearMonth.of(year, month).lengthOfMonth()
+      case (todayMonth, todayYear) =>
+        if((todayYear == year) && (todayMonth == month)) today.getDayOfMonth - 1 else 1
```
Comment (wg-hmrc) [inline app/forms/mappings/SelectLocalDateFormatter.scala:152]: is `if statement` necessary ? since its already handled in the `case statement`

Comment (wg-hmrc) [inline app/forms/mappings/SelectLocalDateFormatter.scala:152]: would `today.getDayOfMonth - 1` not result to 0? which is not a valid day

Comment (magicaljellybeans) [reply app/forms/mappings/SelectLocalDateFormatter.scala:152]: As it is written currently, `case _` is unreachable, so that if else is being used.

Comment (magicaljellybeans) [reply app/forms/mappings/SelectLocalDateFormatter.scala:152]: Maybe just remove the `else 1` and put the `if` as a guard clause before `=>`

File: test/forms/mappings/SelectLocalDateFormatterSpec.scala [hunk 2]
```diff
@@ -152,6 +151,7 @@ class SelectLocalDateFormatterSpec extends SpecBase {
     val month: Int = 12
     val month1: Int = 14
     val day: Int = 1
+    val today: Int = LocalDate.now.getDayOfMonth-1
```
Comment (magicaljellybeans) [inline test/forms/mappings/SelectLocalDateFormatterSpec.scala:154]: Maybe `yesterday` would be a more appropriate name.

Comment (magicaljellybeans) [reply test/forms/mappings/SelectLocalDateFormatterSpec.scala:154]: But also needs to handle the 1st day of the month properly.

File: app/forms/mappings/SelectLocalDateFormatter.scala (hunk 1 above)
Comment (magicaljellybeans) [reply app/forms/mappings/SelectLocalDateFormatter.scala:152]: I have corrected this

File: test/forms/mappings/SelectLocalDateFormatterSpec.scala (hunk 2 above)
Comment (magicaljellybeans) [reply test/forms/mappings/SelectLocalDateFormatterSpec.scala:154]: Sorted this.

**Review by wg-hmrc — APPROVED** [review]
LGTM

**Pull request merged**